File: Python3/0435_non_overlapping_intervals.py
# https://leetcode-cn.com/problems/non-overlapping-intervals/

class Solution:
    def eraseOverlapIntervals(self, intervals):
        """
        Inputs:
            intervals: list[list[int]]
        Outputs:
            res: int
        """

        intervals.sort(key=lambda x:x[1])
        n = len(intervals)

        # Count how many intervals that meet the requirement.
        count = 1
        right = intervals[0][1]
        for i in range(1,n):
            if intervals[i][0] >= right:
                count = count + 1
                right = intervals[i][1]

        return n - count

File: Cpp/0435_non_overlapping_intervals.cpp
// https://leetcode.cn/problems/non-overlapping-intervals/

class Solution {
public:
    int eraseOverlapIntervals(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end(),
            [](const vector<int>& interval1, const vector<int>& interval2) {
                return interval1[1] < interval2[1];
            });
        int cnt = 1, n = intervals.size();
        int right = intervals[0][1];
        for (int i = 1; i < n; ++i) {
            if (intervals[i][0] >= right) {
                cnt += 1;
                right = intervals[i][1];
            }
        }
        int res = n - cnt;
        return res;
    }
};
